### ISSUE TITLE
Error / processing paralogs -> collapse mistranslations

### ISSUE THREAD

**Issue by KasPas22** [gtonkinhill/panaroo#214]
I tried to add new Prokka annotated genomes to my old project and wanted to rerun the panaroo analysis as well. Updated python, prokka and panaroo and created a new conda environment for the project.

However the panaroo pipeline collapses on processing paralogs step with this kind of error message:

Processing paralogs...
100%|█████████████████████████████████████████████████████████████████████████████████████████| 39/39 [00:00<00:00, 371.75it/s]
collapse mistranslations...
Traceback (most recent call last):
  File "/PATH/envs/panaroo/bin/panaroo", line 10, in <module>
    sys.exit(main())
  File "/PATH/miniconda3/envs/panaroo/lib/python3.8/site-packages/panaroo/__main__.py", line 358, in main
    G = collapse_families(G,
  File "/PATH/miniconda3/envs/panaroo/lib/python3.8/site-packages/panaroo/clean_network.py", line 113, in collapse_families
    cdhit_clusters = iterative_cdhit(G,
  File "/PATH/miniconda3/envs/panaroo/lib/python3.8/site-packages/panaroo/cdhit.py", line 411, in iterative_cdhit
    run_cdhit_est(input_file=temp_input_file.name,
  File "/PATH/miniconda3/envs/panaroo/lib/python3.8/site-packages/panaroo/cdhit.py", line 147, in run_cdhit_est
    subprocess.run(cmd, shell=True, check=True)
  File "/PATH/miniconda3/envs/panaroo/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'cd-hit-est -T 1 -i /PATH/tmpwzwciacf/tmp_z9bost1 -o /PATH/tmpwzwciacf/tmpmq9vjj4z -c 0.99 -s 0.0 -aL 0.0 -AL 99999999 -aS 99999999 -AS 99999999 -r 1 -M 0 -d 999 -mask NX -n 7 > /dev/null' returned non-zero exit status 1. 

I'm very thankful for any ideas on the cause of this issue and how to get around it! 

Error message is similar when panaroo is run in other environments of conda and with different data sets. 
biopython                 1.80       
cd-hit                    4.8.1   
panaroo                   1.3.2 
python                    3.8.15
prokka                    1.13

**Comment by iagosblancos** [comment]
Just read your issue now, and I realized having had the same problem here. I made sure to have all dependencies updated just like you, and my error message is similar. Hence, my output folder does not contain all generated files as indicated in Documentation.

**Comment by gtonkinhill** [comment]
Sorry, it looks like I missed the original posting of this issue!

It is hard to diagnose the problem from this error message but I am guessing that there may be a problematic annotation that is missing the filters in Panaroo.

If you are able to reproduce the crash and then run the subsequent cd-hit command it should hopefully give a more informative error message. In the above example, you would run

```
cd-hit-est -T 1 -i /PATH/tmpwzwciacf/tmp_z9bost1 -o /PATH/tmpwzwciacf/tmpmq9vjj4z -c 0.99 -s 0.0 -aL 0.0 -AL 99999999 -aS 99999999 -AS 99999999 -r 1 -M 0 -d 999 -mask NX -n 7
```

If this does not highlight the problem, then if you can send me a small reproducible example that would help a lot in identifying the issue.

**Comment by iagosblancos** [comment]
> Sorry, it looks like I missed the original posting of this issue!
> 
> It is hard to diagnose the problem from this error message but I am guessing that there may be a problematic annotation that is missing the filters in Panaroo.
> 
> If you are able to reproduce the crash and then run the subsequent cd-hit command it should hopefully give a more informative error message. In the above example, you would run
> 
> ```
> cd-hit-est -T 1 -i /PATH/tmpwzwciacf/tmp_z9bost1 -o /PATH/tmpwzwciacf/tmpmq9vjj4z -c 0.99 -s 0.0 -aL 0.0 -AL 99999999 -aS 99999999 -AS 99999999 -r 1 -M 0 -d 999 -mask NX -n 7
> ```
> 
> If this does not highlight the problem, then if you can send me a small reproducible example that would help a lot in identifying the issue.

As I got the same problem, Just tried to reproduce the solution. However, It has prompted in the screen a list of instructions about CD-HIT. My differences compared to KasPas22 are the names of files and folders, but issues are similar.
My command was:
`cd-hit-est -T 18 -i /home/iago/PiBAC assemblies/ncbi-genomes-2022-11-22/gff/resultsgff/tmpdrm4ml_s/tmp8csd5w5i -o /home/iago/PiBAC assemblies/ncbi-genomes-2022-11-22/gff/resultsgff/tmpdrm4ml_s/tmp_36vo7m2 -c 0.99 -s 0.0 -aL 0.0 -AL 99999999 -aS 99999999 -AS 99999999 -r 1 -M 0 -d 999 -mask NX -n 7`

And My output in the screen:
================================================================
Program: CD-HIT, V4.8.1 (+OpenMP), Apr 07 2021, 10:57:21
Command: cd-hit-est -T 18 -i /home/iago/PiBAC
         assemblies/ncbi-genomes-2022-11-22/gff/resultsgff/tmpdrm4ml_s/tmp8csd5w5i
         -o /home/iago/PiBAC
         assemblies/ncbi-genomes-2022-11-22/gff/resultsgff/tmpdrm4ml_s/tmp_36vo7m2
         -c 0.99 -s 0.0 -aL 0.0 -AL 99999999 -aS 99999999 -AS
         99999999 -r 1 -M 0 -d 999 -mask NX -n 7

Started: Sat Feb  4 09:40:19 2023
```
================================================================
                            Output                              
----------------------------------------------------------------
                ====== CD-HIT version 4.8.1 (built on Apr  7 2021) ======

Usage: cd-hit-est [Options] 

Options

   -i   input filename in fasta format, required, can be in .gz format
   -j   input filename in fasta/fastq format for R2 reads if input are paired end (PE) files
         -i R1.fq -j R2.fq -o output_R1 -op output_R2 or
         -i R1.fa -j R2.fa -o output_R1 -op output_R2 
   -o   output filename, required
   -op  output filename for R2 reads if input are paired end (PE) files
   -c   sequence identity threshold, default 0.9
        this is the default cd-hit's "global sequence identity" calculated as:
        number of identical amino acids or bases in alignment
        divided by the full length of the shorter sequence
   -G   use global sequence identity, default 1
        if set to 0, then use local sequence identity, calculated as :
        number of identical amino acids or bases in alignment
        divided by the length of the alignment
        NOTE!!! don't use -G 0 unless you use alignment coverage controls
        see options -aL, -AL, -aS, -AS
   -b   band_width of alignment, default 20
   -M   memory limit (in MB) for the program, default 800; 0 for unlimitted;
   -T   number of threads, default 1; with 0, all CPUs will be used
   -n   word_length, default 10, see user's guide for choosing it
   -l   length of throw_away_sequences, default 10
   -d   length of description in .clstr file, default 20
        if set to 0, it takes the fasta defline and stops at first space
   -s   length difference cutoff, default 0.0
        if set to 0.9, the shorter sequences need to be
        at least 90% length of the representative of the cluster
   -S   length difference cutoff in amino acid, default 999999
        if set to 60, the length difference between the shorter sequences
        and the representative of the cluster can not be bigger than 60
   -aL  alignment coverage for the longer sequence, default 0.0
        if set to 0.9, the alignment must covers 90% of the sequence
   -AL  alignment coverage control for the longer sequence, default 99999999
        if set to 60, and the length of the sequence is 400,
        then the alignment must be >= 340 (400-60) residues
   -aS  alignment coverage for the shorter sequence, default 0.0
        if set to 0.9, the alignment must covers 90% of the sequence
   -AS  alignment coverage control for the shorter sequence, default 99999999
        if set to 60, and the length of the sequence is 400,
        then the alignment must be >= 340 (400-60) residues
   -A   minimal alignment coverage control for the both sequences, default 0
        alignment must cover >= this value for both sequences 
   -uL  maximum unmatched percentage for the longer sequence, default 1.0
        if set to 0.1, the unmatched region (excluding leading and tailing gaps)
        must not be more than 10% of the sequence
   -uS  maximum unmatched percentage for the shorter sequence, default 1.0
        if set to 0.1, the unmatched region (excluding leading and tailing gaps)
        must not be more than 10% of the sequence
   -U   maximum unmatched length, default 99999999
        if set to 10, the unmatched region (excluding leading and tailing gaps)
        must not be more than 10 bases
   -B   1 or 0, default 0, by default, sequences are stored in RAM
        if set to 1, sequence are stored on hard drive
        !! No longer supported !!
   -P   input paired end (PE) reads, default 0, single file
        if set to 1, please use -i R1 -j R2 to input both PE files
   -cx  length to keep after trimming the tail of sequence, default 0, not trimming
        if set to 50, the program only uses the first 50 letters of input sequence
   -cy  length to keep after trimming the tail of R2 sequence, default 0, not trimming
        if set to 50, the program only uses the first 50 letters of input R2 sequence
        e.g. -cx 100 -cy 80 for paired end reads
   -ap  alignment position constrains,  default 0, no constrain
        if set to 1, the program will force sequences to align at beginings
        when set to 1, the program only does +/+ alignment
   -p   1 or 0, default 0
        if set to 1, print alignment overlap in .clstr file
   -g   1 or 0, default 0
        by cd-hit's default algorithm, a sequence is clustered to the first 
        cluster that meet the threshold (fast cluster). If set to 1, the program
        will cluster it into the most similar cluster that meet the threshold
        (accurate but slow mode)
        but either 1 or 0 won't change the representatives of final clusters
   -r   1 or 0, default 1, by default do both +/+ & +/- alignments
        if set to 0, only +/+ strand alignment
   -mask        masking letters (e.g. -mask NX, to mask out both 'N' and 'X')
   -match       matching score, default 2 (1 for T-U and N-N)
   -mismatch    mismatching score, default -2
   -gap gap opening score, default -6
   -gap-ext     gap extension score, default -1
   -bak write backup cluster file (1 or 0, default 0)
   -sc  sort clusters by size (number of sequences), default 0, output clusters by decreasing length
        if set to 1, output clusters by decreasing size
   -sf  sort fasta/fastq by cluster size (number of sequences), default 0, no sorting
        if set to 1, output sequences by decreasing cluster size
        this can be very slow if the input is in .gz format
   -h   print this help

   Questions, bugs, contact Weizhong Li at liwz@sdsc.edu
   For updated versions and information, please visit: http://cd-hit.org
                                                    or https://github.com/weizhongli/cdhit

   cd-hit web server is also available from http://cd-hit.org

   If you find cd-hit useful, please kindly cite:

   "CD-HIT: a fast program for clustering and comparing large sets of protein or nucleotide sequences", Weizhong Li & Adam Godzik. Bioinformatics, (2006) 22:1658-1659
   "CD-HIT: accelerated for clustering the next generation sequencing data", Limin Fu, Beifang Niu, Zhengwei Zhu, Sitao Wu & Weizhong Li. Bioinformatics, (2012) 28:3150-3152
```

**Comment by gtonkinhill** [comment]
Hi,

Can you post the command you used to run Panaroo? I've just noticed the input files reported seem a little odd.

**Comment by iagosblancos** [comment]
Hello, my command to use Panaroo was:
`panaroo -i *.gff -o resultsgff/ --clean-mode strict -a core -t 18`

In this folder, I got some dozens of gff files from Prokka.

**Comment by gtonkinhill** [comment]
Hi,

I'm afraid I have not been able to reproduce this issue on my end. From your output, it isn't obvious to me why CD-HIT is not running. 

If you are able to send through a small reproducible example, I would be happy to take a look.

**Comment by iagosblancos** [comment]
Hello, just leaving a recent update of the issue!
I've realized my PC was having problems with conda, then I decided to reinstall Panaroo and its dependencies manually instead of using conda. Before this, I was troubled in "collapsing mistranslations", which led to bugs. Right now, its working, after manual installation according to the instructions on:
https://gtonkinhill.github.io/panaroo/#/gettingstarted/installation
